docs/index.rst

```diff
@@ -10,5 +10,11 @@ Documentation content:
    :maxdepth: 1
 
    User guide <userguide/index>
+   build_meta
+   pkg_resources
+   references/keywords
+   roadmap
+   setuptools
```

Comment:
I've linked the orphans here but they'll need to be cleaned up separately. I think it's out of scope of this PR.

cc @alvyjudy